docs/extended_api/synchronization_library/pipeline/producer_acquire.md

```diff
@@ -0,0 +1,17 @@
+---
+nav_exclude: true
+---
+
+# cuda::pipeline\<Scope>::**producer_acquire**
+
+```c++
+void producer_acquire();
+```
+
+Blocks the current thread until the next _pipeline stage_ is available.

```

Comment:
Do we document the relationship between `producer_acquire` and `cuda::pipeline_shared_state`'s `StagesCount` template parameter somewhere?

If not, I think we should.

Reply:
For example, we guarantee that here:

```c++
__global__ void foo() {
constexpr unsigned stages_count = 2;
__shared__ cuda::pipeline_shared_state<cuda::thread_scope::thread_scope_block, stages_count> shared_state;
￼auto pipeline = cuda::make_pipeline(group, &shared_state);
pipeline.producer_acquire();
...
pipeline.producer_commit();
pipeline.producer_acquire(); // This acquire never blocks
...
```

the second call to `producer_acquire` never blocks because the `pipeline` has a "capacity` for holding two Stages.

If we change the `StageCount` to `1`: 

```c++
__global__ void foo() {
constexpr unsigned stages_count = 1;
__shared__ cuda::pipeline_shared_state<cuda::thread_scope::thread_scope_block, stages_count> shared_state;
￼auto pipeline = cuda::make_pipeline(group, &shared_state);
pipeline.producer_acquire();
...
pipeline.producer_commit();
pipeline.producer_acquire(); // This acquire always blocks, deadlocks, ... ??
...
```

The second `producer_acquire` call cannot ever make progress AFAICT. And that's something that we also guarantee.

Not sure where would be the right place to document this, but I think it should be in the libcudacxx docs.

Reply:
Yes, let's get this change in, we can iterate on top of it after.

Reply:
Created #77 for this.